src/components/user/login/index.js

```diff
@@ -0,0 +1,82 @@
+import React, { Component } from 'react';
+import {
+  Grid, Row, Col, Alert,
+} from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+import { loginAPI } from '../../../store/actions/login';
+import { ROUTES } from '../../../constants';
+import LoginForm from './login-form';
+
+class Login extends Component {
+  constructor() {
+    super();
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(fields) {
+    const { login } = this.props;
+    login(fields);
+  }
+
+  renderFormError() {
+    const { loginState } = this.props;
+    const { message } = loginState;
+
+    return message ? <Alert bsStyle="danger">{message}</Alert> : null;
+  }
+
+  render() {
+    const { loginState } = this.props;
+    const { statusCode, data } = loginState;
+    const { home } = ROUTES;
+    const isLoginSuccessful = statusCode === 200;
+    const loginFormProps = {
+      title: 'Login',
+      onSubmit: this.handleSubmit,
+      inputFields: [
+        { textInput: { name: 'email' } },
+        { textInput: { name: 'password' } },
+      ],
+    };
+
+    if (isLoginSuccessful) {
+      localStorage.setItem('token', data);
```

Comment:
Logic to setting the auth token should be handled in the reducer, not in the component. Remember the `render` method is fired every time a prop or state value changes. That means, with the current login you'll be unnecessarily storing the token multiple times on each of those re-renders, which is an anti-pattern as it degrades the application's performance.

Reply:
Ah you are right. I didnt think about that one. Sorry I haven't done any react in a while but I thought it only re-renders when the state is updated or if shouldComponentUpdate() returns true. I never knew that it re-renders when props changes... that's new for me.

Reply:
On second thought... I do not see how I am storing the token "multiple times". As soon as I store the token in the localstorage, I am navigating to the home page which means that there will no be re-render right after I've stored the token. There will no be cases where the application stores the token again unless an user navigates to the login page again and re-login. I will update the code nevertheless